va/va_test.go

```diff
@@ -226,6 +217,54 @@ func tlssniSrvWithNames(t *testing.T, chall core.Challenge, names ...string) *ht
 	return hs
 }
 
+func tlsalpn01Srv(t *testing.T, chall core.Challenge, names ...string) *httptest.Server {
+	template := tlsCertTemplate(names)
+	certBytes, _ := x509.CreateCertificate(rand.Reader, template, template, &TheKey.PublicKey, &TheKey)
+	cert := &tls.Certificate{
+		Certificate: [][]byte{certBytes},
+		PrivateKey:  &TheKey,
+	}
+
+	shasum := sha256.Sum256([]byte(chall.ProvidedKeyAuthorization))
+	acmeExtension := pkix.Extension{
+		Id:       IdPeAcmeIdentifierV1,
+		Critical: true,
+		Value:    shasum[:],
+	}
+
+	template.ExtraExtensions = []pkix.Extension{acmeExtension}
+	certBytes, _ = x509.CreateCertificate(rand.Reader, template, template, &TheKey.PublicKey, &TheKey)
+	acmeCert := &tls.Certificate{
+		Certificate: [][]byte{certBytes},
+		PrivateKey:  &TheKey,
+	}
+
+	tlsConfig := &tls.Config{
+		Certificates: []tls.Certificate{},
+		ClientAuth:   tls.NoClientCert,
+		GetCertificate: func(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
+			if clientHello.ServerName != names[0] {
+				return nil, nil
+			}
+			if len(clientHello.SupportedProtos) == 1 && clientHello.SupportedProtos[0] == ACMETLS1Protocol {
+				return acmeCert, nil
+			}
+			return cert, nil
+		},
+		NextProtos: []string{"http/1.1", ACMETLS1Protocol},
+	}
+
+	hs := httptest.NewUnstartedServer(http.DefaultServeMux)
+	hs.TLS = tlsConfig
+	hs.Config.TLSNextProto = map[string]func(*http.Server, *tls.Conn, http.Handler){
+		ACMETLS1Protocol: func(_ *http.Server, conn *tls.Conn, _ http.Handler) {
+			conn.Close()
```

Comment:
The error from this needs to be caught or one of our linting tests will complain, easiest is to just `_ = conn.Close()` since we don't really care about any error return here...